documentation/docs/01-routing.md

```diff
@@ -161,6 +161,45 @@ The `body` property of the request object will be provided in the case of POST r
 - Form data (with content-type `application/x-www-form-urlencoded` or `multipart/form-data`) will be parsed to a read-only version of the [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData) object.
 - All other data will be provided as a `Uint8Array`
 
+#### HTTP Method Overrides
+
+Given that the only valid `<form>` methods are GET and POST, functionality is provided to override this limitation and allow the use of other HTTP verbs. This is particularly helpful when ensuring your application works even when javascript fails or is disabled.
+
+- Disabled by default to prevent unintended behavior for those who don't need it
+- For security purposes, the original method on the form must be `POST` and cannot be overridden with `GET`
```

Comment:
this is a bit vague. it'd be better if we can say something more specific
```suggestion
- To protect against CSRF attacks, the original method on the form must be `POST` and cannot be overridden with `GET`
```

Reply:
Agreed. Although I'm not sure if this is for CSRF protection or for protection against other attack vectors (cache poisoning was the example given: https://github.com/sveltejs/kit/issues/1046#issuecomment-949382948).

Unfortunately I lack expertise in this field. Maybe @Prinzhorn could approve your suggestion or provide a more detailed/accurate statement?

Reply:
I don't think we need to be that specific and it's also not just a security thing. GET has vastly different semantics from POST, PUT etc. We should focus on why the feature exist, what problem it solves. I don't see any problem being solved by changing the method from/to GET.

I'd shorten the paragraph and the two bullet points to something like:

---

In contrast to `fetch` the only valid methods for a `<form>` are `GET` and `POST`. Svelte allows you to override the `<form>` method to workaround this limitation if you need to. That way your application can transparently work when JavaScript fails or is disabled by using `fetch` and `<form>` interchangeably with the same endpoint.

Reply:
I like your version of why this feature exists but I'm unsure about removing those 2 bullet points. It should be pointed out that the feature needs to be enabled by the user. The defaults are noted later in the configuration section but they might not notice that. And explaining that the form method must be POST and cannot be overridden with GET will prevent users from using this feature incorrectly and forcing them to figure out what is wrong when they receive build errors or their app doesn't function as expected.

Reply:
I agree about documenting everything. But for me docs are a last resort (the actual text, anything beyond skimming through code examples) when things already went wrong and I need to figure out why. But we can do much better. During dev when we see `_method` and `methodOverride` is disabled we can _tell_ the developer that it needs to be enabled. Same if we see `_method` with something other than `POST`. Same for all other cases that are currently silently ignored. If the method is not in `allowedMethods` arguable it should even return a 400 in production. Or at the very least during dev it should _tell_ you. We have the knowledge, let's not make the user run into unexpected behavior (e.g. silently ignoring `_method` for `GET`). Instead let's fail as loud as possible so they don't need to open their browser to actually read the docs or search though `/issues`. Let them stay in the zone and be like "oh, I need to set `enabled: true`, gotcha, thanks friendly error message".

Reply:
Agreed. Can anyone tell me how compile/build warnings are generated in Sveltekit?

And I'm on the fence regarding returning a 400 if the method isn't in allowed methods, but I agree silently ignoring it is probably not the correct action. I'd like to handle it similar to other scenarios in Sveltekit.